apps/emqx_authn/src/emqx_authn_api.erl

```diff
@@ -576,7 +578,11 @@ authenticator(delete, #{bindings := #{id := AuthenticatorID}}) ->
     delete_authenticator([authentication], ?GLOBAL, AuthenticatorID).
 
 authenticator_status(get, #{bindings := #{id := AuthenticatorID}}) ->
-    lookup_from_all_nodes(?GLOBAL, AuthenticatorID).
+    with_authenticator(
+        AuthenticatorID,
+        [authentication],
+        fun(_) -> lookup_from_all_nodes(?GLOBAL, AuthenticatorID) end
```

Comment:
Not sure if this code is equivalent.
Before we'd try to lookup authenticator on each node in the cluster, now we only lookup on other nodes when this authenticator is found in the configuration on the local node?
Or did I get it wrong?

Reply:
We don't do it instead, we do it in addition. I explained why in the comment above. Alternative would be to also do a remote lookup in `/authentication/{name}` and then (for both resources) return `404` if just a single one of the remote calls returns a `not_found`?

Reply:
I mentioned it in the ticket but I think the design of the API here is flawed, it should include node names, and maybe have an explicit `all` node for aggregates. Do we want to leak node names though? How does the cluster behave if nodes have different configurations? Is this by design and a wanted situation? If not, who should be in charge to resolve it (the API or the 'system' itself)?